core/src/provider/initiator.rs

```diff
@@ -200,13 +198,6 @@ where
 		self
 	}
 
-	/// Overwrites the used keystore pointer that will be used when initiating the
-	/// structs for a core builder.
-	pub fn with_keystore(&mut self, keystore: SyncCryptoStorePtr) -> &mut Self {
-		self.keystore = Some(keystore);
-		self
-	}
```

Comment:
This must not be removed. The keystore is now inner part of the `ExeutionExtension` struct

Reply:
Yes, but that's why we now have `with_exec_extensions`. One can provide the keystore and/or strategies this way.

Alternatively, we could also add:

```
	pub fn with_keystore(&mut self, keystore: SyncCryptoStorePtr) -> &mut Self {
		let ext = ExecutionExtensions::new(ExecutionStrategies::default(), Some(keystore), None);
		self.with_exec_extensions(ext)
	}

	pub fn with_exec_strategies(&mut self, strategies: ExecutionStrategies) -> &mut Self {
		let ext = ExecutionExtensions::new(strategies, None, None);
		self.with_exec_extensions(ext)
	}
```

WDYT?